### PR TITLE
added pfSense Parsers for syslog-ng and Logstash

### PR DESCRIPTION
**Pull request by noodlemctwoodle**
Fixes #

## Proposed Changes
- Added pfSense parser for Syslog-ng
- Added pfSense parser for Logstash (Local Traffic)
- Added pfSense parser for Logstash (GeoIP Traffic)

### CODE REVIEW

**Review by noodlemctwoodle — COMMENTED** [review]
added file extensions `.txt`

**Review by shainw — CHANGES_REQUESTED** [review]
These should have some further details included and have a couple example rows of data.  Please see the parser guidance here - https://github.com/Azure/Azure-Sentinel/wiki/Contribute-to-Sentinel-GitHub-Community-of-Queries#example-parser

**Comment by shainw** [comment]
@noodlemctwoodle - please see my comments.  Apologies on the delay, have been heads down on a recent security incident.

**Comment by shainw** [comment]
@noodlemctwoodle - Please remember to update the parsers with the proper format and sample data.  If not, we can close this for now.  Thanks, Shain

**Comment by shainw** [comment]
@noodlemctwoodle - Checking in again to make sure you saw the request for changes.

**Comment by noodlemctwoodle** [comment]
Sorry I've been super busy, I'll make these changes tomorrow 😂

Apologies for the delay

**Comment by shainw** [comment]
@noodlemctwoodle - Understood, I know exactly where you are coming from. :) Thanks!! Shain.

**Comment by noodlemctwoodle** [comment]
@shainw Good Evening 

I have updated the headers for the parsing queries. 

Thanks

File: Parsers/pfSense/syslogng-pfsense.txt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+//
+// Syslog-NG Parsing Rule
+// https://docs.netgate.com/pfsense/en/latest/monitoring/logs/raw-filter-format.html​​​​​​​
+//
+// Usage Instruction :
+// Extract rule descriptions with associated tracking number (Optional & pfSense Only)
+// In pfSense and go to diagnostics -> Command Prompt
+// Enter one of the following command in the execute shell command box and click the execute button
+//
+// pfctl -vv -sr | grep label | sed -r 's/@([[:digit:]]+).*(label "|label "USER_RULE: )(.*)".*/"\1","\3"/g' | sort -V -u | awk 'NR==1{$0="\"Rule\",\"Label\""RS$0}7'
```
Comment (shainw) [inline Parsers/pfSense/syslogng-pfsense.txt:12]: Is this a sample event that would show up in the table?  It looks like a command line to execute.  We need an event sample and I don't see any Sample data for pfSense - see step 4 from here - https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/ReadMe.md#build-the-connector.

File: Parsers/pfSense/logstash-pfsense-local.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// LogStash Instructions:
+// pfSense/opnSense - Logstash Configuration - https://github.com/noodlemctwoodle/pf-azure-sentinel
+//
+// Parser Notes:
+// This parser will parse pfSense or opnSense LAN logs and tag them with GeoIP data using MaxMind db or Logstash GeoIP db.
+//
+// Usage Instruction :
+// Paste below query in log analytics, click on Save button and select as Function from drop down by specifying function name and alias (e.g. pf_GeoIPDestination) or
+// paste in the command into your Azure Sentinel logs and run the query.
+//
```
Comment (shainw) [inline Parsers/pfSense/logstash-pfsense-local.txt:13]: We need an event sample and I don't see any Sample data for pfSense - see step 4 from here - https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/ReadMe.md#build-the-connector.

File: Parsers/pfSense/logstash-pfsense-geoIp.txt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+// pfSense/opnSense - Logstash Configuration - https://github.com/noodlemctwoodle/pf-azure-sentinel
+//
+// Parser Notes:
+// This parser will parse pfSense or opnSense WAN logs and tag them with GeoIP data using MaxMind db or Logstash GeoIP db.
+//
+// Usage Instruction :
+// Paste below query in log analytics, click on Save button and select as Function from drop down by specifying function name and alias (e.g. pf_GeoIPDestination) or
+// paste in the command into your Azure Sentinel logs and run the query.
+//
+//
```
Comment (shainw) [inline Parsers/pfSense/logstash-pfsense-geoIp.txt:14]: We need an event sample and I don't see any Sample data for pfSense - see step 4 from here - https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/ReadMe.md#build-the-connector.

**Review by shainw — CHANGES_REQUESTED** [review]

**Comment by shainw** [comment]
@noodlemctwoodle - reminder this still needs some changes.

**Comment by noodlemctwoodle** [comment]
Hi @shainw

Thanks I am aware, I've just had a major outage to deal with as my virtual host boot disk failed last weekend, so I've been having to rebuild. I should be back up and running by this weekend.

I can then rebuild my Logstash server and import my data back into Sentinel and get the screenshots 😃

File: Parsers/pfSense/logstash-pfsense-geoIp.txt (hunk 3 above)
Comment (noodlemctwoodle) [reply Parsers/pfSense/logstash-pfsense-geoIp.txt:14]: @shainw Added screenshot

File: Parsers/pfSense/logstash-pfsense-local.txt (hunk 2 above)
Comment (noodlemctwoodle) [reply Parsers/pfSense/logstash-pfsense-local.txt:13]: @shainw - Added screenshot

File: Parsers/pfSense/logstash-pfsense-geoIp.txt (hunk 3 above)
Comment (shainw) [reply Parsers/pfSense/logstash-pfsense-geoIp.txt:14]: @noodlemctwoodle - Sorry if not clear, we need an actual sample event in a string format as would show up in an export of the row of data into a CSV or the like, then we can test the parser.

**Review by shainw — CHANGES_REQUESTED** [review]

**Comment by shainw** [comment]
Very old, closing for now, please feel free to re-open when ready to continue work.

**Pull request closed without merging**